internal/statemachine/common_states.go

```diff
@@ -415,8 +415,16 @@ func (stateMachine *StateMachine) makeDisk() error {
 		if !found {
 			continue
 		}
-		imgName := filepath.Join(stateMachine.commonFlags.OutputDir, stateMachine.VolumeNames[volumeName])
 
+		// eMMC volumes should not have its own disk image, these are treated specially
+		// as they consist only of fixed content images that needs to be written during
+		// runtime (i.e gadget install or gadget update). The contents of the image are
+		// part of the gadget snap
+		if volume.Schema == "emmc" {
```

Comment:
suggestion: define a `schemaEMMC = "emmc"` const to avoid using this string in multiple places.